src/app/resize/resize.component.ts

```diff
@@ -0,0 +1,72 @@
+import {Component, ElementRef, EventEmitter, Input, Output} from "@angular/core";
+
+@Component({
+	selector: 'resize',
+	templateUrl: `
+		<div [class.overlay]="isExpanded"></div>
+		<div
+			class="handle"
+			[class.is-expanded]="isExpanded">
+		</div>
+		`,
+	styles: [`
+	.handle {
+		position: absolute;
+		right: 0;
+		top: 0;
+		height: 100%;
+		width: 15px;
+		border-right: 1px dashed rgba(128, 128, 128, 0.5);
+		cursor: col-resize;
+	}
+
+	.is-expanded {
+		width: 100%;
+	}
+
+	.overlay {
+		position: fixed;
+		top: 0;
+		right: 0;
+		left: 0;
+		bottom: 0;
+		width: 100%;
+		height: 100%;
+	}
+  `],
+	host: {
+		'(mousemove)': 'onMouseMove($event)',
```

Comment:
What would be the benefit of listening this way as opposed to using @HostListener (I'm glad there's no github user with this name, haha)?